jujugui/static/gui/src/app/store/env/bakery.js

```diff
@@ -18,721 +18,535 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
-            data, null, null, false, null, onAuthDone);
-        }.bind(this);
-        return this.webhandler.sendPutRequest(
-          path, this._prepareHeaders({'Content-type': JSON_CONTENT_TYPE}),
-          data, null, null, false, null,
-          this._requestHandlerWithInteraction.bind(
-            this, onAuthRequired, onAuthDone, failureCallback, redirect)
-        );
-      },
-
-      /**
-       Takes the path supplied by the caller and makes a patch request to the
-       requestHandlerWithInteraction instance. If setCookiePath is set then
-       it is used to set a cookie back to the ui after authentication.
-
-       @param path {String} The path to make the api request to.
-       @param data {String} Stringified JSON of parameters to send to the PATCH
-              endpoint.
-       @param successCallback {Function} Called when the api request completes
-              successfully.
-       @param failureCallback {Function} Called when the api request fails
-              with a response of >= 400 except 401/407 where it does
-              authentication.
-       @param {Boolean} redirect Whether the handler should redirect if there
-              is a 401 on the request.
-       @param {Object} response The XHR response object from initial request.
-      */
-      sendPatchRequest: function(
-        path, data, successCallback, failureCallback, redirect) {
-        const onAuthDone = this._requestHandler.bind(
-          this, successCallback, failureCallback);
-        const onAuthRequired = function() {
-          return this.webhandler.sendPatchRequest(
-            path, this._prepareHeaders({'Content-type': JSON_CONTENT_TYPE}),
-            data, null, null, false, null, onAuthDone);
-        }.bind(this);
-        return this.webhandler.sendPatchRequest(
-          path, this._prepareHeaders({'Content-type': JSON_CONTENT_TYPE}),
-          data, null, null, false, null,
-          this._requestHandlerWithInteraction.bind(
-            this, onAuthRequired, onAuthDone, failureCallback, redirect)
+      }
+    }
+
+    /**
+      Send an HTTP request to the given URL with the given HTTP method, headers
+      and body. The given callback receives an error and a response when the
+      request is complete.
+
+      @param {String} url The URL to which to send the request.
+      @param {String} method The HTTP method, like "get" or "POST".
+      @param {Object} headers Headers that must be included in the request.
+        Note that bakery specific headers are automatically added internally.
+      @param {String} body The request body if it applies, or null.
+      @param {Function} callback A function called when the response is
+        received from the remote URL. It receives a tuple (error, response).
+        If the request succeeds the error is null.
+      @return {Object} the XHR instance.
+    */
+    sendRequest(url, method, headers, body, callback) {
```

Comment:
There are a number of positional arguments here, and I see that you're assigning the values to variables in methods below so that the call signatures make sense (yay!). BUT maybe this should instead take a single args object and callback so that we don't have to do that?